test/tools/yulopti.cpp

```diff
@@ -87,37 +89,70 @@ class YulOpti
 		}.printErrorInformation(_errors);
 	}
 
-	void parse(string const& _input)
+	shared_ptr<Object> getSubObject(shared_ptr<Object> const& _object, string const& _qualifiedPath)
```

Comment:
From what I understand, the purpose of this function is to be given a top level object, and then based on the `_qualifiedPath` to find the 'sub object' recursively; however, is it possible for the top level input object to be the the one that we've specified via the `_qualifiedPath` parameter? If yes, then this function should be called `getObject`. 

In any case, it would be good to document this function to mention that it searches for the sub object recursively based on the `_qualifiedPath` parameter.